pkg/chartutil/create.go

```diff
@@ -511,13 +512,15 @@ spec:
 var Stderr io.Writer = os.Stderr
 
 // CreateFrom creates a new chart, but scaffolds it from the src chart.
-func CreateFrom(chartfile *chart.Metadata, dest, src string) error {
+func CreateFrom(name, dest, src string, keepMetadata bool) error {
```

Comment:
Helm follows Semantic Versioning in the public Go API. That means that function signatures cannot be changed until we increment the major version of Helm.

Please see the [`NewShow`](https://github.com/helm/helm/blob/a27c468c82fe2cb9a4e07d7c82c9df6b6fed9ed7/pkg/action/show.go#L66) and `NewShowWithConfig` functions for an example of creating a new function and deprecating an old one to handle a signature.

We have to maintain backward compatibility in the API.

Reply:
Hi,

Thanks for your feedback! :slightly_smiling_face: 

I added another function (like `NewShowWithConfig`) which gets called from `CreateFrom` with `keepMetadata` set to `false` so that it preserves the old behavior when using the old function.

Please see https://github.com/helm/helm/pull/11273/files#diff-cb9c038056214a9f3a39d58d63fb15b872c6057381c0926d75929b9dda01adf8R514-R523

Is it okay like this?

Reply:
I just noticed that my first approach does break backwards compatibility by ignoring the user-provided `chart.Metadata` parameter (except its `name` field).

https://github.com/helm/helm/pull/11273/commits/ce6e1265b42ea7b918d09de4fa85fcec64f949e3 should fix this by replacing the `bool` variable with a `*chart.Metadata` pointer which gets set to `nil` if the user sets the `--no-override` flag. Without `--no-override` the output should be exactly the same as before (Chart.yaml without comments).

Here's the relevant function: https://github.com/krakowski/helm/blob/feat/keep-metadata/pkg/chartutil/create.go#L701-L734